shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -451,53 +480,80 @@ class JSBackend(allowUnresolvedSymbols: Boolean) {
     JSClassDecl(classSymbol.runtimeName, fields, base, members, traits)
   }
 
-  protected def translateMixinDeclaration(
-      mixinSymbol: MixinSymbol
-  )(implicit scope: Scope): JSClassMethod = {
-    val getterScope = scope.derive(s"getter ${mixinSymbol.lexicalName}")
-    val mixinScope = getterScope.derive(s"mixin ${mixinSymbol.lexicalName}")
-    // Collect class fields.
-    val fields = mixinSymbol.body.collectFields ++
-      mixinSymbol.body.collectTypeNames.flatMap(resolveTraitFields)
-
-    mixinSymbol.methods.foreach {
-      case MethodDef(_, _, Var(nme), _, _) => mixinScope.declareNewClassMember(nme, N, true)
+  protected def translateSelfDeclaration(self: ValueSymbol): Ls[JSStmt] =
+    if (visitedSymbols(self)) {
+      visitedSymbols -= self
+      JSConstDecl(self.runtimeName, JSIdent("this")) :: Nil
     }
-    mixinSymbol.ctor.foreach {
-      case NuFunDef(rec, Var(nme), _, _) => mixinScope.declareNewClassMember(nme, rec, false)
-      case _ => ()
+    else Nil
+
+  protected def translateLocalNewType(typeDef: NuTypeDef)(implicit scope: Scope): JSConstDecl = {
+    // TODO: traitSymbols?
```

Comment:
Assert they're empty.

Reply:
Should I need to add test cases to see if the `throw` works fine?

Reply:
No, this is fine.